jbake-core/src/main/java/org/jbake/parser/MarkupEngine.java

```diff
@@ -253,33 +253,35 @@ private void processDefaultHeader(ParserContext context) {
             if (hasHeaderSeparator(line)) {
                 break;
             }
-            processLine(line, context.getDocumentModel());
+            processHeaderLine(line, context.getDocumentModel());
         }
     }
 
-    private void processLine(String line, Map<String, Object> content) {
+    private void processHeaderLine(String line, Map<String, Object> content) {
         String[] parts = line.split("=", 2);
         if (!line.isEmpty() && parts.length == 2) {
+            processHeader(parts[0], parts[1], content);
+        }
+    }
 
-
-            String key = sanitizeKey(parts[0]);
-            String value = sanitizeValue(parts[1]);
-
-            if (key.equalsIgnoreCase(Crawler.Attributes.DATE)) {
-                DateFormat df = new SimpleDateFormat(configuration.getDateFormat());
-                try {
-                    Date date = df.parse(value);
-                    content.put(key, date);
-                } catch (ParseException e) {
-                    LOGGER.error("unable to parse date {}", value);
-                }
-            } else if (key.equalsIgnoreCase(Crawler.Attributes.TAGS)) {
-                content.put(key, getTags(value));
-            } else if (isJson(value)) {
-                content.put(key, JSONValue.parse(value));
-            } else {
-                content.put(key, value);
+    void processHeader(String inputKey, String inputValue, Map<String, Object> content) {
```

Comment:
I'm searching for a better name here. How about `processHeaderValue`

Reply:
I'm ok with changing the name, I used `processHeader` because this is key/value, an alternative would be `storeHeaderValue`, but I have no strong feeling into any direction.

Reply:
Thats even a better name. That alignes with my attempt to build a document and template model abstraction. Which could be a new home for this method later. See #532